dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -633,6 +633,14 @@ void MergeTreeData::setTTLExpressions(const ColumnsDescription::ColumnTTLs & new
 
     if (new_ttl_table_ast)
     {
+        std::unique_lock<std::mutex> move_ttl_entries_lock;
+        if (!only_check)
```

Comment:
Need to minimize critical section. Let's take lock only when we update class fields.

Reply:
+